File: leetcode/2914-minimum-number-of-changes-to-make-binary-string-beautiful/solution.cpp
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful
 * Runtime: 0ms (100.00%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    int minChanges(string s) {
        int len = s.size();
        int count = 0;

        for (int i = 0; i < len - 1; i += 2)
            if (s[i] != s[i + 1])
                count++;

        return count;
    }
};


/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful
 * Runtime: 4ms (30.85%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    int minChanges(string s) {
        int len = s.size();
        int count = 0;
        int i = 0;

        while (i < len - 1) {
            if (s[i] != s[i + 1])
                count++;
            i += 2;
        }

        return count;
    }
};

File: leetcode/2914-minimum-number-of-changes-to-make-binary-string-beautiful/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful
* Runtime: 15ms (97.30%)
"""

class Solution:
    def minChanges(self, s: str) -> int:
        count: int = 0

        for i in range(0, len(s) - 1, 2):
            if s[i] != s[i + 1]:
                count += 1

        return count

"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem: https://leetcode.com/problems/minimum-number-of-changes-to-make-binary-string-beautiful
* Runtime: 27ms (29.73%)
"""

class Solution:
    def minChanges(self, s: str) -> int:
        count: int = 0
        i: int = 0

        while i < len(s) - 1:
            if s[i] != s[i + 1]:
                count += 1
            i += 2

        return count